lightning/src/ln/channelmanager.rs

```diff
@@ -5293,6 +5496,19 @@ where
 	R::Target: Router,
 	L::Target: Logger,
 {
+	/// Returns `MessageSendEvent`s strictly ordered per-peer, in the order they were generated.
+	/// The returned array will contain `MessageSendEvent`s for different peers if
+	/// `MessageSendEvent`s to more than one peer exists, but `MessageSendEvent`s to the same peer
+	/// is always placed next to each other.
+	///
+	/// Note that that while `MessageSendEvent`s are strictly ordered per-peer, the peer order for
+	/// the chunks of `MessageSendEvent`s for different peers is random. I.e. if the array contains
+	/// `MessageSendEvent`s  for both `node_a` and `node_b`, the `MessageSendEvent`s for `node_a`
+	/// will randomly be placed first or last in the returned array.
+	///
+	/// Note that even though `BroadcastChannelAnnouncement` and `BroadcastChannelUpdate`
+	/// `MessageSendEvent`s are intended to be broadcasted to all peers, they will be pleaced among
+	/// the `MessageSendEvent`s to the specific peer they were generated under.
 	fn get_and_clear_pending_msg_events(&self) -> Vec<MessageSendEvent> {
```

Comment:
Let me know if you'd like me to change this function to return a `HashMap<PublicKey, Vec<MessageSendEvent>>` instead now that the peer order is random, as it's probably more intuitive and less prune to errors when handling the messages,.
Opted to keep the interface as it is to start with, to minimize the changes, but will change if you think it's worth it in this PR.

Reply:
Ah yeah good point, let's keep it as a plain `Vec` for now then :)!

Maybe I should explain why I think the `Hashmap` option is less prune to errors though, if it's worth considering :).
Imagine that someone writes a test in with `no-std` configuration activated, which yields two `MessageSendEvent` to two different peers. If the `per_peer_state` `HashMap` iteration order isn't randomized in `no-std`, then the message order will be the same  from `get_and_clear_pending_msg_events` when writing the test.   I think it's then quite unintuitive that you need to call the `remove_first_msg_event_to_node` when checking the messages in the test, so maybe there's a probability that people won't.
It will then be really confusing for people why their tests then fail 50% of the time in `std` configuration, when the message order is actually randomized.

Given that risk, maybe it would be worth writing a `test` configuration version of `get_and_clear_pending_msg_events` that yields the messages with a different interface than the `non-test` version, or that perhaps sorts the peer's before pushing the messages to the `Vec` or something?